File: 프로그래머스/python/LV3/이중우선순위큐.py
import heapq


def solution(operations):
    answer = []
    q = []
    for i in operations:
        order, num = i.split(" ")
        num = int(num)
        if order == "I":
            heapq.heappush(q, num)
        else:
            if num == -1 and len(q):
                heapq.heappop(q)
            elif num == 1 and len(q):
                q.sort()
                q.pop()

    return [0, 0] if not len(q) else [max(q), min(q)]

File: 프로그래머스/javascript/LV3/이중우선순위큐.js
function solution(operations) {
  let arr = [];

  for (const op of operations) {
    const [char, num] = op.split(" ");
    if (char === "I") {
      arr.push(Number(num));
      arr.sort((a, b) => a - b);
    } else {
      if (arr.length === 0) continue;
      if (num === "-1") {
        arr.shift(); // 최솟값 제거
      } else {
        arr.pop(); // 최댓값 제거
      }
    }
  }

  if (arr.length === 0) return [0, 0];
  return [arr[arr.length - 1], arr[0]];
}
